op-service/hardforks/forks.go

```diff
@@ -0,0 +1,17 @@
+package hardforks
+
+//go:generate go run ./gen/main.go -type=Fork -input=./forks.go
+type Fork uint32
```

Comment:
I like the code-gen quality, but I think the approach encourages features to go into the main-line of hardforks too much.
Ideally we model it such that feature-flags can be added without implied hardfork schedule. I.e. some forks are not sequential.

I think the ideal form, to encourage feature-flag like grouping, would be to implement a json-unmarshal routine that can parse the existing fork-schedule, but then load it into a more dynamic one, where it's a map of feature-flags to activation points, and a map of forks to included features.